tests/test_policy_eval.py

```diff
@@ -8,57 +8,114 @@ def assert_pe_results(spec, answers):
 
     for i, pi in enumerate(spec['Pi_phi']):
         pe = PolicyEval(amdp)
-        res = pe.run(pi, True)
+        results = pe.run(pi, True)
 
-        for j, res in enumerate(res):
-            assert(np.allclose(answers[i][j], res))
+        for k in answers.keys():
+            for j, res in enumerate(results):
+                assert(np.allclose(answers[k][i][j], res[k]))
 
 def test_example_3():
     spec = load_spec('example_3')
-    answers = [[
-        np.array([3.8125, 4.5, 0.75, 0, 0, 0, 0]),
-        np.array([3.8125, 3.5625, 0]),
-        np.array([3.8125, 3.5625, 0]),
-    ]]
+    answers = {
+        'v':[[
+            np.array([3.8125, 4.5, 0.75, 0]), # mdp
+            np.array([3.8125, 3.5625, 0]),    # amdp / mc*
+            np.array([3.8125, 3.5625, 0]),    # td
+        ]],
+        'q': [[
+            np.array([[4.5, 1.75],      # mdp
+                      [5,   3   ],
+                      [0,   3   ],
+                      [0,   0   ]]),
+            np.array([[4.5 , 1.75],     # amdp
+                      [3.75, 3  ],
+                      [0   , 0  ]]),
+            np.array([[3.5625, 4.5625], # td
+                      [3.75  , 3     ],
+                      [0     , 0     ]])
+        ]]
+    }
 
     assert_pe_results(spec, answers)
 
 def test_example_11():
     spec = load_spec('example_11')
-    answers = [[
-        np.array([0.14285714, 0.28571429, 0.57142857, 0]),
-        np.array([0.14285714, 0.42857143, 0]),
-        np.array([0.2, 0.4, 0]),
-    ]]
+    answers = {
+        'v':[[
+            np.array([0.14285714, 0.28571429, 0.57142857, 0]),
+            np.array([0.14285714, 0.42857143, 0]),
```

Comment:
I vote to make these fractions: `1/7`, `2/7`, etc.

Reply:
Done